backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts

```diff
@@ -28,11 +36,41 @@ const interceptor = aws4Interceptor({
 axiosInstance.interceptors.request.use(interceptor);
 
 describe("POST /token", () => {
-  describe("Given there is no grant type", () => {
+  let clientIdAndSecret: string;
+
+  beforeAll(async () => {
+    const clientDetails = await getFirstClientDetails();
+    clientIdAndSecret = `${clientDetails.client_id}:${clientDetails.client_secret}`;
+  });
+
+  describe("Given there is no grant_type in the request body", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/token`,
+        "",
+        {
+          headers: { "x-custom-auth": "Basic " + toBase64(clientIdAndSecret) },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_grant",
+        error_description: "Invalid grant type or grant type not specified",
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given the grant_type value is invalid", () => {
```

Comment:
I'm not sure if these first two tests are both required as they return exactly the same response body.

Reply:
I'm happy for these tests to serve as documentation and regression

Reply:
Ok, keeping both then... thanks!